fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/dao/EraseResourceDAO.java

```diff
@@ -0,0 +1,351 @@
+import com.ibm.fhir.persistence.jdbc.dao.api.IResourceReferenceDAO;
+import com.ibm.fhir.persistence.jdbc.dao.impl.ResourceDAOImpl;
+
+/**
+ * EraseDAO is the data access layer of the erase operation which executes directly
+ * against the database using SQL statements to:
+ * <li>check the resource exists</li>
+ * <li>delete all versions from the resource table</li>
+ * <li>delete all parameters</li>
+ * <li>delete entries from global table</li>
+ * <li>delete resource specific logical resource entry</li>
+ *
+ * @implNote the first version of this class used a ThreadPool from OpenLiberty,
+ * in the Runnable, the timeout was managed through a Parameter passed in from the client.
+ * This used a ManagedConnection (see WSRdbManagedConnectionImpl.java in OpenLiberty) and a new threadid,
+ * even with a passed JEEContext,SecurityContext and BaseContext in the ThreadPool
+ * Postgres forks a new Connection and activates a new UserTransaction (with one already running).
+ * This is not ideal, however, to manage this we've implemented a version specific erase.
+ * If a timeout is hit (however unlikely), the user is expected to erase each version, and then erase the whole thing.
+ */
+public class EraseResourceDAO extends ResourceDAOImpl {
+
+    private static final String CLASSNAME = EraseResourceDAO.class.getSimpleName();
+    private static final Logger LOG = Logger.getLogger(CLASSNAME);
+
+    private static final String CALL_POSTGRES = "{CALL %s.ERASE_RESOURCE(?, ?, ?)}";
+    private static final String CALL_DB2 = "CALL %s.ERASE_RESOURCE(?, ?, ?)";
+
+    // The translator specific to the database type we're working with
+    private final IDatabaseTranslator translator;
+
+    private ResourceEraseRecord eraseRecord;
+    private EraseDTO eraseDto;
+    private Connection conn;
+
+    /**
+     * Public constructor
+     *
+     * @param connection
+     * @param translator
+     * @param schemaName
+     * @param flavor
+     * @param cache
+     * @param rrd
+     */
+    public EraseResourceDAO(Connection conn, IDatabaseTranslator translator, String schemaName, FHIRDbFlavor flavor, FHIRPersistenceJDBCCache cache,
+            IResourceReferenceDAO rrd) {
+        super(conn, schemaName, flavor, cache, rrd);
+        this.translator = translator;
+        this.conn = getConnection();
+    }
+
+    /**
+     * Execute the stored procedure/function to erase the content.
+     *
+     * @param callStr the sql that should be executed
+     * @throws Exception
+     */
+    private void runCallableStatement(String callStr) throws Exception {
+        try (CallableStatement call = conn.prepareCall(String.format(callStr, getSchemaName()));){
+            call.setString(1, eraseDto.getResourceType());
+            call.setString(2, eraseDto.getLogicalId());
+            call.registerOutParameter(3, Types.BIGINT);
+            call.execute();
+
+            Integer deleted = (int) call.getLong(3);
```

Comment:
No need to use Integer here because it can't be null. Assumes that the procedure will never return a null result, of course.

Reply:
absolutely correct, you are find cruft from the evolution of the code.  I've fixed locally. It'll be in the next commit.